src/components/ripple/ripple.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Renderer,
+  SimpleChange,
+  ViewEncapsulation,
+} from '@angular/core';
+
+enum RippleState {
+  NEW,
+  EXPANDING,
+  FADING_OUT,
+}
+
+class Ripple {
+  state = RippleState.NEW;
+  constructor(public element: Element) {}
+}
+
+const RIPPLE_SPEED_PX_PER_SECOND = 500;
+const MIN_RIPPLE_FILL_TIME_SECONDS = 0.2;
+const MAX_RIPPLE_FILL_TIME_SECONDS = 0.6;
+
+const sqr = (x: number) => x * x;
+
+const distanceToFurthestCorner = (x: number, y: number, parentRect: any) => {
+  const x1 = parentRect.left;
+  const x2 = x1 + parentRect.width;
+  const y1 = parentRect.top;
+  const y2 = y1 + parentRect.height;
+  const maxSquaredDistance = Math.max(
+      sqr(x - x1) + sqr(y - y1),
+      sqr(x2 - x) + sqr(y - y1),
+      sqr(x - x1) + sqr(y2 - y),
+      sqr(x2 - x) + sqr(y2 - y));
+  return Math.sqrt(maxSquaredDistance);
+};
+
+
+@Component({
+  moduleId: module.id,
+  template: `<div class="md-ripple-background"></div>`,
+  selector: 'md-ink-ripple',
+  styleUrls: ['ripple.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdInkRipple implements OnInit, OnDestroy, OnChanges {
+  /**
+   * The element that triggers the ripple when click events are received. Defaults to the parent
+   * of the <md-ink-rippple>.
+   */
+  @Input('trigger') _trigger: Element;
+  /**
+   * If true, the ripple always originates from the center of the <md-ink-ripple> bounds rather
+   * than originating from the location of the click event.
+   */
+  @Input('centered') _centered: boolean;
+  /**
+   * If true, click events will not trigger the ripple. It can still be triggered by manually
+   * calling start() and end().
+   */
+  @Input('disabled') _disabled: boolean;
+  /**
+   * Custom color for ripples.
+   */
+  @Input('color') _color: string;
+  /**
+   * Custom color for the ripple background.
+   */
+  @Input('backgroundColor') _backgroundColor: string;
+  /**
+   * If set, the normal duration of ripple animations is divided by this value. For example,
+   * setting it to 0.5 will cause the animations to take twice as long.
+   */
+  @Input('speedFactor') _speedFactor: number = 1;
```

Comment:
Why are all of these inputs private? I think it would be fine making them public.
(also they're missing the actual `private` keyword)